src/forms/NumberInput.jsx

```diff
@@ -0,0 +1,184 @@
+import React from 'react';
+import cx from 'classnames';
+import Flex from '../layout/Flex';
+import FlexItem from '../layout/FlexItem';
+import Icon from '../media/Icon';
+
+export const DECREMENT_BTN_CLASS = 'decrementButton';
+export const INCREMENT_BTN_CLASS = 'incrementButton';
+
+/**
+ * @module NumberInput
+ */
+class NumberInput extends React.Component {
+	constructor(props) {
+		super(props);
+		this.state = {
+			value: props.value || '',
+		};
+
+		this.decrement = this.decrement.bind(this);
+		this.increment = this.increment.bind(this);
+		this.onBlur = this.onBlur.bind(this);
+		this.onChange = this.onChange.bind(this);
+		this.onFocus = this.onFocus.bind(this);
+	}
+
+	onBlur(e) {
+		if (document.activeElement !== this.fauxInputEl && document.activeElement !== this.decrementBtnEl && document.activeElement !== this.incrementBtnEl) {
+			this.fauxInputEl.classList.remove('focused');
+		}
+	}
+
+	onChange(e) {
+		this.setState({ value: e.target.value });
+		if (this.props.onChange) {
+			this.props.onChange(e);
+		}
+	}
+
+	onFocus(e) {
+		this.fauxInputEl.classList.add('focused');
+	}
+
+	increment() {
+		const currentVal = new Number(this.state.value);
+		const increment = new Number(this.props.step);
+		const newValue = currentVal + increment;
+
+		if (newValue > this.props.max) return;
+
+		this.setState({value: newValue});
+	}
+
+	decrement() {
+		const currentVal = new Number(this.state.value);
+		const decrement = new Number(this.props.step);
+		const newValue = currentVal - decrement;
+
+		if (newValue < this.props.min) return;
+
+		this.setState({value: newValue});
+	}
+
+	render() {
+		const {
+			children,
+			className,
+			error,
+			id,
+			label,
+			labelClassName,
+			max,
+			min,
+			name,
+			onBlur, // eslint-disable-line no-unused-vars
+			onFocus, // eslint-disable-line no-unused-vars
+			onChange, // eslint-disable-line no-unused-vars
+			required,
+			step,
+			value, // eslint-disable-line no-unused-vars
+			...other
+		} = this.props;
+
+		const classNames = {
+			field: cx(
+				'field--reset',
+				{ 'field--error': error },
+				className
+			),
+			label: cx(
+				{ required },
+				labelClassName
+			),
+			incrementBtn: cx(
+				'button--reset',
+				INCREMENT_BTN_CLASS
+			),
+			decrementBtn: cx(
+				'button--reset',
+				DECREMENT_BTN_CLASS
+			)
+		};
+
+		return (
+			<div>
+				<label className={classNames.label} htmlFor={id}>
+					{label}
+				</label>
+
+				<div
+					className='fauxInput'
+					ref={ el => this.fauxInputEl = el }>
+					<Flex align='center'>
+						<FlexItem>
+							<input type='number'
+								name={name}
+								max={max}
+								min={min}
+								step={step}
+								value={this.state.value}
+								required={required}
+								className={classNames.field}
+								onBlur={this.onBlur}
+								onFocus={this.onFocus}
+								onChange={this.onChange}
+								{...other} />
+						</FlexItem>
+
+						<FlexItem shrink>
+							<button
```

Comment:
I try and only use `Button` for things that I want to look like a button. The `Button` component adds button classes and styling that I don't want to have to fight here, even with the `reset` prop.
If you think there's something to be gained by using the `Button` component, let me know and I'll gladly switch over.

PS: I've already [filed a task](https://meetup.atlassian.net/browse/SDS-280) to make `Button reset` to reset more styles.